core/controllers/topics_and_skills_dashboard.py

```diff
@@ -60,12 +60,21 @@ def get(self):
         skill_summary_dicts = [
             summary.to_dict() for summary in skill_summaries]
 
+        triaged_skill_ids = topic_services.get_all_triaged_skills()
         topic_rights_dict = topic_services.get_all_topic_rights()
         for topic_summary in topic_summary_dicts:
             topic_rights = topic_rights_dict[topic_summary['id']]
             if topic_rights:
                 topic_summary['is_published'] = (
                     topic_rights.topic_is_published)
+                topic_summary['can_edit_topic'] = (
+                    topic_services.check_can_edit_topic(self.user, topic_rights)
+                )
+
+        untriaged_skill_summary_dicts = []
+        for skill_summary in skill_summary_dicts:
```

Comment:
This is O(N^2) and N can be quite large. Maybe triaged_skill_ids should be a dict instead for O(1) lookup.

Reply:
Shall I use sets?  since the value field for such a dict is arbitrary and I am already using sets in get_all_triaged_skills() in topic_services. I will remove the conversion to list which I am doing now, before returning.

Reply:
That seems fine, just analyze complexity and make sure it's no higher than O(N log N).